merlin/models/tf/transforms/bias.py

```diff
@@ -206,6 +212,39 @@ def update(
     def compute_output_shape(self, input_shape):
         return input_shape
 
+    def call(
+        self, outputs: Union[Prediction, PredictionOutput], features, training=False, **kwargs
+    ):
+        if training:
+            if not isinstance(outputs, Prediction):
+                # The PredictionTask classes (V1) are
+                # applying the log-q correction in `call_outputs`
+                return outputs
+
+            positive_item_ids = tf.squeeze(features[self.candidate_id_name])
+            negative_item_ids = outputs.negative_candidate_ids
+            positive_probs = tf.gather(self.candidate_probs, positive_item_ids)
+
+            if negative_item_ids is not None:
+                negative_probs = tf.gather(self.candidate_probs, negative_item_ids)
+                # repeat negative scores for each positive item
+                negative_probs = tf.reshape(
+                    tf.tile(tf.squeeze(negative_probs), tf.shape(positive_item_ids)[0:1]),
+                    (-1, tf.shape(negative_item_ids)[0]),
+                )
+                positive_probs = tf.concat(
+                    [tf.expand_dims(positive_probs, -1), negative_probs], axis=1
+                )
+
+            # Applies the logQ correction
+            epsilon = 1e-16
+            predictions = outputs.predictions - (
```

Comment:
we could move this shared logic into a separate method since it's now being used by both `call` and `call_outputs`? (everything between `postiive_probs = ...` and this line)

then this would become something like:

```python
predictions = self.compute_log_q_correction(predictions, positive_item_ids, negative_item_ids)
```